src/network.rs

```diff
@@ -54,13 +54,27 @@ pub struct TileDownloader {
     pub retries: usize,
     pub retry_delay: Duration,
     pub tile_storage_folder: Option<PathBuf>,
+    pub zoom_format: String,
 }
 
 impl TileDownloader {
     pub async fn download_tile(
         &self,
-        tile_reference: TileReference,
+        mut tile_reference: TileReference,
     ) -> Result<Tile, TileDownloadError> {
+        if !"auto".eq(&self.zoom_format) {
```

Comment:
This is probably not the right place to handle that. This option is specific to the iiif dezoomer, isn't it? If it is, then it should be handled in the dezoomer.